app/controllers/v1/events_controller.rb

```diff
@@ -0,0 +1,11 @@
+module V1
+  class EventsController < ApplicationController
+    def index
```

Comment:
This implementation was a bit confusing to me. It could be cleaned up if the `.bag` scope were implemented as a filter that did nothing if not given a bag_id.  Absent that, perhaps this more clearly shows the intention:

```ruby
      events = if params[:bag_id]
        Event.bag(params[:bag_id])
      else
        Event
      end
      @events = policy_scope(events)
```